### PR TITLE
audio-player-ui start

### PR DESCRIPTION
**Pull request by Javier-Szyfer**
Starting to work on this proof-of-concept for the audio player ui.

Will right some notes here soon on future steps

### CODE REVIEW

**Comment by Javier-Szyfer** [comment]
This PR is only for documenting progress -  still doesn't working as intended, but getting there soon

**Comment by salieflewis** [comment]
1. I believe react-icons needs to be added as a dev dependency.
2. Upon first loading the page, if you try to mute using the slider, things will break because 'e.target.value is undefined'

https://user-images.githubusercontent.com/37086973/191847701-020d15e2-b36c-4dc8-be55-e761d071f840.mp4

File: examples/nextjs/src/components/PageWrapper.tsx [hunk 1]
```diff
@@ -1,42 +1,72 @@
...
       <Header />
       <main {...props} className="p-4">
+        <AudioPlayer
+          id={INIT_TEST_MUSIC_NFT.id}
+          artist={INIT_TEST_MUSIC_NFT.artist}
+          title={INIT_TEST_MUSIC_NFT.title}
+          image={INIT_TEST_MUSIC_NFT.image}
+          audioSrc={INIT_TEST_MUSIC_NFT.audioSrc}
+        />
+        <audio
```
Comment (dblodorn) [inline examples/nextjs/src/components/PageWrapper.tsx:54]: The audio element should be in the package component.
That way it's completely portable and does not require the user to do a bunch of additional setup on their end.

We could eventually structure the component in a way that allows a user to optionally pass in an external audio tag - or object or whatever but that would take some extra consideration.

File: packages/audio-player-ui/src/components/player-content/controls/PlayToggle/PlayToggle.tsx [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
+import useAudioPlayer from "../../../../hooks/useAudioPlayer";
 
-export const PlayToggle = ({ playing, setPlaying }: any) => {
+export const PlayToggle = () => {
+  const { togglePlay, playing } = useAudioPlayer();
   return (
     <>
       <button
-        onClick={() => setPlaying(!playing)}
+        onClick={togglePlay}
```
Comment (dblodorn) [inline packages/audio-player-ui/src/components/player-content/controls/PlayToggle/PlayToggle.tsx:9]: If you memoize these handlers using `useCallback` they will be way less buggy.

File: packages/audio-player-ui/src/components/player-content/wrappers/PlayerWrapper/PlayerWrapper.tsx [hunk 3]
```diff
@@ -3,8 +3,6 @@ export function PlayerWrapper({
 }: {
   children?: JSX.Element | JSX.Element[];
```
Comment (dblodorn) [inline packages/audio-player-ui/src/components/player-content/wrappers/PlayerWrapper/PlayerWrapper.tsx:4]: never thought of using `JSX.Element[]` 😏

File: packages/audio-player-ui/src/hooks/useAudioPlayer.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    null
+  );
+  const [duration, setDuration] = useState(0);
+  const [curTime, setCurTime] = useState(0);
+  const [playing, setPlaying] = useState(false);
+  const [clickedTime, setClickedTime] = useState(0);
+  const [muted, setMuted] = useState(false);
+  console.log("currentTrack", currentTrack);
+
+  const toggleMute = () => {
```
Comment (dblodorn) [inline packages/audio-player-ui/src/hooks/useAudioPlayer.ts:14]: We should memoize all the handler functions with react useCallback
https://dmitripavlutin.com/dont-overuse-react-usecallback/

example from similar set of handlers:
https://github.com/dblodorn/blocksyncer/blob/main/%40media-player/useMediaPlayer.tsx

borrow as much as you like - and or - fork this repo and mess around with the @media-player directory. It was written to eventually be a plugin.

**Comment by dblodorn** [comment]
@Javier-Szyfer 

When passing in refs to a handler - good to check if it is available first:
https://github.com/dblodorn/blocksyncer/blob/main/%40media-player/useMediaPlayer.tsx#L10-L19

**Comment by dblodorn** [comment]
@Javier-Szyfer Here's a whole chunk of code for the audioplayer from
[https://embed.zora.co/](https://embed.zora.co/)
[Example with audio player](https://embed.zora.co/config/0xf68cc00f5bf70bd39542d2e2dee3f718733408d9/2)

https://gist.github.com/dblodorn/67c1b90abbbadc6dd13383f271a4480a

**Pull request closed without merging**